### PR TITLE
feat: Remove scaling restriction

### PR DESCRIPTION
**Pull request by DanielArndt**
# Description

This change removes the scaling restriction on the `manual-tls-certificates` charm.

This charm is workloadless, and can operate fine with scaling. That being said, the `tls-certificates-interface` that it depends on seems to hardcode some restrictions to the leader unit. This means that all actions need to be performed on the leader unit.

I'm honestly not sure what the best approach is. This was the quickest, but we could also update the `tls-certificates-interface` to support non-leader units.

I fear that there may be some edge cases around when the charm is scaled relative to other operations.

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/integration/test_integration.py [hunk 1]
```diff
@@ -128,9 +141,10 @@ async def test_given_requirer_requests_certificate_creation_when_deploy_then_sta
             timeout=1000,
         )
 
-    async def test_given_tls_requirer_is_deployed_and_related_when_provide_certificate_then_certificate_is_passed_correctly(  # noqa: E501
-        self, ops_test, charm, cleanup
+    async def test_given_tls_requirer_is_deployed_with_3_units_and_related_when_provide_certificate_then_certificate_is_passed_correctly(  # noqa: E501
```
Comment (DanielArndt) [inline tests/integration/test_integration.py:144]: I wasn't quite sure if this should be a separate test...

File: README.md [hunk 2]
```diff
@@ -18,12 +18,14 @@ If the optional parameter relation-id is provided then only the information of t
 
 The following action will return all certificate requests that don't have certificates already provided, along with further information (relation-id, application_name and unit_name)
 
+NOTE: If you happen to scale `manual-tls-certificates`, you must run actions on the leader unit. However, there is currently no benefit to scaling this charm.
+
 ```bash
-juju run manual-tls-certificates/leader get-outstanding-certificate-requests relation-id=<id>
+juju run manual-tls-certificates/leader get-outstanding-certificate-requests relation-id=${relation_id}
```
Comment (ghislainbourgeois) [inline README.md:24]: I would revert this change, to keep the consistency with line 31. The previous version also highlights better when viewed on GitHub.

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (DanielArndt) [reply README.md]: I really despise this format (and "highlights better" is subjective, since it is parsing it incorrectly :laughing:) -- but I suppose that's a discussion for another time.

Removed in https://github.com/canonical/manual-tls-certificates-operator/pull/81/commits/9c3a4d8ec796bb06c3b0de64fc22304ebf0113c7

**Review by ghislainbourgeois — APPROVED** [review]
